platform/commonUI/general/test/controllers/DateTimePickerControllerSpec.js

```diff
@@ -57,6 +65,119 @@ define(
+                };
+                mockScope.time = {
+                    hours: 12,
+                    minutes: 34,
+                    seconds: 56
+                };
+                fireWatchCollection("date", mockScope.date);
+                expect(mockScope.ngModel[mockScope.field])
+                    .toEqual(moment.utc("1998-01-06 12:34:56").valueOf());
+            });
+
+            describe("once initialized with model state", function () {
+                var testTime = moment.utc("1998-01-06 12:34:56").valueOf();
+
+                beforeEach(function () {
+                    fireWatch("ngModel[field]", testTime);
+                });
+
+                it("exposes date/time values in scope", function () {
+                    expect(mockScope.date.year).toEqual(1998);
+                    expect(mockScope.date.month).toEqual(0); // Months are zero-indexed
+                    expect(mockScope.date.day).toEqual(6);
+                    expect(mockScope.time.hours).toEqual(12);
+                    expect(mockScope.time.minutes).toEqual(34);
+                    expect(mockScope.time.seconds).toEqual(56);
+                });
+
+                it("provides names for time properties", function () {
+                    Object.keys(mockScope.time).forEach(function (key) {
+                        expect(mockScope.nameFor(key))
+                            .toEqual(jasmine.any(String));
+                    });
+                });
+
+                it("provides options for time properties", function () {
+                    Object.keys(mockScope.time).forEach(function (key) {
+                        expect(mockScope.optionsFor(key))
+                            .toEqual(jasmine.any(Array));
+                    });
+                });
+
+                it("exposes times to populate calendar as a table", function () {
+                    var matchingCell;
+                    // Should be able to find the selected date
+                    mockScope.table.forEach(function (row) {
+                        row.forEach(function (cell) {
+                            if (cell.dayOfYear === 6) {
+                                matchingCell = cell;
+                            }
+                        });
+                    });
+                    expect(matchingCell).toEqual({
+                        year: 1998,
+                        month: 0,
+                        day: 6,
+                        dayOfYear: 6
+                    });
+                });
+
+                it("allows the displayed month to be advanced", function () {
+                    // Around the edges of the displayed calendar we
+                    // may be in previous or subsequent month, so
+                    // test around the middle.
+                    var i, originalMonth = mockScope.table[2][0].month;
+
+                    function mod12(month) {
+                        return ((month % 12) + 12) % 12;
```

Comment:
JavaScript's `%` is a [remainder instead of a modulo](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder_()); for positive values these are equivalent but for negative values they are not. Expected behavior is modulo-like (months should rollover and remain non-negative)